### PR TITLE
unix: add os information

### PR DESCRIPTION
**Pull request by mmalecki**
This still needs Windows version.

/cc @bnoordhuis

### CODE REVIEW

File: src/unix/os.c [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ */
+
+#include "uv.h"
+#include "internal.h"
+
+#include <errno.h>
+#include <string.h>
+#include <sys/utsname.h>
+
+uv_err_t uv_hostname(char* hostname, size_t size) {
```
Comment (bnoordhuis) [inline src/unix/os.c]: Style: two newlines between functions and headers / declarations.

File: src/unix/os.c [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  int r = gethostname(hostname, size);
```
Comment (bnoordhuis) [inline src/unix/os.c]: Style: split the declaration and the assignment.

File: src/unix/os.c [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+  if (r < 0) {
+    return uv__new_sys_error(errno);
+  }
```
Comment (bnoordhuis) [inline src/unix/os.c]: In general, single line if statements don't have braces, i.e.:

``` c
if (r == -1)
  return uv__new_sys_error(errno);
```

File: src/unix/os.c [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  struct utsname info;
+
+  int r = uname(&info);
+
+  if (r < 0) {
+    return uv__new_sys_error(errno);
+  }
+
+  strncpy(type, info.sysname, size - 1);
+  type[size] = '\0';
```
Comment (bnoordhuis) [inline src/unix/os.c]: Use uv_strlcpy() here.

File: test/test-platform-output.c [hunk 5]
```diff
@@ -83,5 +83,17 @@
   }
   uv_free_interface_addresses(interfaces, count);
 
+  err = uv_hostname(buffer, 512);
```
Comment (bnoordhuis) [inline test/test-platform-output.c]: s/512/sizeof(buffer)/

File: include/uv.h [hunk 6]
```diff
@@ -1467,6 +1467,10 @@ UV_EXTERN uv_err_t uv_interface_addresses(uv_interface_address_t** addresses,
 UV_EXTERN void uv_free_interface_addresses(uv_interface_address_t* addresses,
   int count);
 
+UV_EXTERN uv_err_t uv_hostname(char* hostname, size_t size);
+UV_EXTERN uv_err_t uv_os_type(char* type, size_t size);
+UV_EXTERN uv_err_t uv_os_release(char* release, size_t size);
```
Comment (bnoordhuis) [inline include/uv.h]: Can you add some short comments describing what the function does and what inputs it takes / outputs it produces?

**Comment by mmalecki** [comment]
@bnoordhuis all fixed.

**Comment by bnoordhuis** [comment]
LGTM but it needs Windows stubs (bonus points for working implementations). I suggest that you copy out an empty string besides returning UV_ENOSYS.

**Comment by bnoordhuis** [comment]
@mmalecki Are you still planning on finishing this? It'll need some rebasing to apply to master.

**Comment by bnoordhuis** [comment]
No followup, closing.

**Pull request closed without merging**